### ISSUE TITLE
Enable cygwin builds on gh-actions

### ISSUE THREAD

**Issue by james-d-mitchell** [libsemigroups/libsemigroups#281]
At some point recently the appveyor CI started failing because of an issue with perl in cygwin when trying to run `./autogen.sh`: 

~~~~
Can't locate threads.pm in @INC (you may need to install the threads module) (@INC contains: /usr/share/automake-1.16 /usr/local/lib/perl5/site_perl/5.32/i686-cygwin-threads-64int /usr/local/share/perl5/site_perl/5.32 /usr/lib/perl5/vendor_perl/5.32/i686-cygwin-threads-64int /usr/share/perl5/vendor_perl/5.32 /usr/lib/perl5/5.32/i686-cygwin-threads-64int /usr/share/perl5/5.32) at /usr/share/automake-1.16/Automake/ChannelDefs.pm line 23.
BEGIN failed--compilation aborted at /usr/share/automake-1.16/Automake/ChannelDefs.pm line 26.
Compilation failed in require at /usr/share/automake-1.16/Automake/Configure_ac.pm line 27.
BEGIN failed--compilation aborted at /usr/share/automake-1.16/Automake/Configure_ac.pm line 27.
Compilation failed in require at /usr/bin/aclocal-1.16 line 38.
BEGIN failed--compilation aborted at /usr/bin/aclocal-1.16 line 38.
autoreconf-2.69: aclocal failed with exit status: 2
~~~~

This might be an issue with the way that the appveyor job is set up or something, but I wasn't able to resolve this so decided for the time being at least to disable appveyor for `libsemigroups`.

**Comment by james-d-mitchell** [comment]
On reflection maybe there's no point in reenabling this.

**Comment by slel** [comment]
GH actions can be set up for Cygwin.

**Comment by james-d-mitchell** [comment]
Thanks @slel that's a good idea, perhaps I'll re-open this and change the title!

**Comment by james-d-mitchell** [comment]
resolved in v2.1.4
